add checkpoint/restore to the API

tych0 opened the pull request:
This adds support for the new checkpoint/restore API functions.

Commit:
add checkpoint/restore to the API

Signed-off-by: Tycho Andersen <tycho.andersen@canonical.com>

caglar10ur commented on lxc.c:
```diff
@@ -304,3 +304,11 @@ bool go_lxc_remove_device_node(struct lxc_container *c, const char *src_path, co
 bool go_lxc_rename(struct lxc_container *c, const char *newname) {
 	return c->rename(c, newname);
 }
+
```
I think it would be better if we include lxc/version.h and decorate those two with something like below

#if LXC_VERSION_MAJOR >= 1 && LXC_VERSION_MINOR >= 1
....
#endif

otherwise one cannot use stable-1.0 with the go bindings. 

Alternatively we can branch again and declare v3 of bindings LXC 1.1+ compatible.

caglar10ur commented on container.go:
```diff
@@ -1353,3 +1353,25 @@ func (c *Container) RemoveDeviceNode(source string, destination ...string) error
 	return nil
 }
+
+func (c *Container) Checkpoint(directory string, stop bool, verbose bool) error {
```
I believe I would prefer passing an options struct called CRUIOptions (or something like that, see https://github.com/lxc/go-lxc/blob/v2/options.go) to those functions like the rest as it allow us to introduce new options or change semantics without breaking the API. 

What do you think?

caglar10ur commented:
Thanks Tycho! I added some comments. If you don't have time I can work on those changes as well ~next week [I'm on vacation right now :)]

tych0 commented:
On Sun, Nov 02, 2014 at 07:19:18AM -0800, S.Çağlar Onur wrote:

> Thanks Tycho! I added some comments. If you don't have time I can work on those changes as well ~next week [I'm on vacation right now :)]

Sure thing! Your suggestions sound good, I will update ASAP. I am also
traveling this week, but I am happy to make the changes.

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/go-lxc/pull/27#issuecomment-61410577

Commit:
Guard c/r with API version test

These are only available with lxc >= 1.1, so we should only include them if
we're building against that api version.

Commit:
Use options struct vs. lots of options for c/r methods

caglar10ur commented:
Hey Tycho,

First of all sorry for delay in my response, I was away from my computer for a while. Overall I have two more comments (sorry for nit picking). Would it be possible to move CheckpointOpts/RestoreOpts into options.go and rename them to CheckpointOptions/RestoreOptions?

If you don't have time I can merge this as is and then push that changes to the branch afterwards (assuming no objections :P)

tych0 commented:
On Wed, Nov 12, 2014 at 12:21:18PM -0800, S.Çağlar Onur wrote:

> Hey Tycho,
> 
> First of all sorry for delay in my response, I was away from my computer for a while. Overall I have two more comments (sorry for nit picking). Would it be possible to move CheckpointOpts/RestoreOpts into options.go and rename them to CheckpointOptions/RestoreOptions?
> 
> If you don't have time I can merge this as is and then push that changes to the branch afterwards (assuming no objections :P)

Sure no problem on either. I will try to push them later today.

> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/lxc/go-lxc/pull/27#issuecomment-62786622

caglar10ur commented:
I merged this and then pushed the minor changes on top of it
